cli/command/environment.go

```diff
@@ -217,42 +231,47 @@ func (e *EnvironmentCommand) update(c *cli.Context) error {
 }
 
 func (e *EnvironmentCommand) link(c *cli.Context) error {
-	args, err := extractArgs(c.Args(), "SOURCE_ENVIRONMENT_NAME", "DESTINATION_ENVIRONMENT_NAME")
-	if err != nil {
-		return err
-	}
+	generateAddLinkRequest := func(sourceEnvironmentID, destEnvironmentID string) (models.UpdateEnvironmentRequest, error) {
+		env, err := e.client.ReadEnvironment(sourceEnvironmentID)
+		if err != nil {
+			return models.UpdateEnvironmentRequest{}, err
+		}
 
-	sourceEnvironmentID, err := e.resolveSingleEntityIDHelper("environment", args["SOURCE_ENVIRONMENT_NAME"])
-	if err != nil {
-		return err
-	}
+		if !client.Contains(destEnvironmentID, env.Links) {
+			appendedLinks := append(env.Links, destEnvironmentID)
+		}
 
-	destEnvironmentID, err := e.resolveSingleEntityIDHelper("environment", args["DESTINATION_ENVIRONMENT_NAME"])
-	if err != nil {
-		return err
+		req := models.UpdateEnvironmentRequest{Links: &appendedLinks}
+		return req, nil
 	}
 
-	if sourceEnvironmentID == destEnvironmentID {
-		return fmt.Errorf("Cannot link an environment to itself")
-	}
+	return e.updateEnvironmentLinksHelper(c, generateAddLinkRequest)
+}
 
-	req := models.CreateEnvironmentLinkRequest{
-		SourceEnvironmentID: sourceEnvironmentID,
-		DestEnvironmentID:   destEnvironmentID,
-	}
+func (e *EnvironmentCommand) unlink(c *cli.Context) error {
+	generateRemoveLinkRequest := func(sourceEnvironmentID, destEnvironmentID string) (models.UpdateEnvironmentRequest, error) {
+		env, err := e.client.ReadEnvironment(sourceEnvironmentID)
+		if err != nil {
+			return models.UpdateEnvironmentRequest{}, err
+		}
 
-	jobID, err := e.client.CreateLink(req)
-	if err != nil {
-		return err
+		updatedLinks := []string{}
+		for _, link := range env.Links {
+			if link != destEnvironmentID {
+				updatedLinks = append(updatedLinks, link)
+			}
+		}
+
+		req := models.UpdateEnvironmentRequest{Links: &updatedLinks}
+
+		return req, nil
 	}
 
-	return e.waitOnJobHelper(c, jobID, "linking", func(environmentID string) error {
-		e.printer.Printf("Environment successfully linked\n")
-		return nil
-	})
+	return e.updateEnvironmentLinksHelper(c, generateRemoveLinkRequest)
 }
 
-func (e *EnvironmentCommand) unlink(c *cli.Context) error {
+func (e *EnvironmentCommand) updateEnvironmentLinksHelper(c *cli.Context,
```

Comment:
when we do line wrapping with params, give each param a line please:
```
func foo(
    p1 int,
    p2 bool,
) error {
    ....
}
```

the last line is especially important as it gives a clear indentation difference between the return signature and body

Reply:
that looks soo much better! ✅